application/common/models/Reset.php

```diff
@@ -487,23 +484,23 @@ public function setType($type, $methodId = null)
         if (in_array($type, [self::TYPE_SPOUSE, self::TYPE_SUPERVISOR, self::TYPE_PRIMARY])) {
             $this->type = $type;
             $this->method_id = null;
-        } elseif ($type == self::TYPE_METHOD) {
+        } elseif (in_array($type, [self::TYPE_METHOD, Method::TYPE_EMAIL, Method::TYPE_PHONE])) {
```

Comment:
Are we intentionally comparing the new Reset `$type` both to a constant in the Reset class and to constants in the Method class? Something seems odd about that...